### PR TITLE
SDIT-2282 Update version of VISITOR_RESTRICTION-UPSERTED

### PR DESCRIPTION
**Pull request by andymarke**
Only 2 events are upserts so a bit more messy which involves a small amount of renaming and better telemetry

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/contactperson/ContactPersonSynchronisationService.kt [hunk 1]
```diff
@@ -77,11 +83,15 @@ class ContactPersonSynchronisationService(
...
+            "$telemetrySuffix-created-success",
+            telemetry,
+          )
         }
-        telemetryClient.trackEvent(
-          "contactperson-person-restriction-synchronisation-created-success",
-          telemetry,
-        )
+      } catch (e: Exception) {
+        telemetry["error"] = e.message.toString()
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/contactperson/ContactPersonSynchronisationService.kt:92]: helpful to easily find error scenarios

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/contactperson/ContactPersonSynchronisationService.kt [hunk 2]
```diff
@@ -50,23 +52,27 @@ class ContactPersonSynchronisationService(
...
-          val nomisRestriction = nomisPerson.restrictions.find { it.id == event.visitorRestrictionId } ?: throw IllegalStateException("Restriction ${event.visitorRestrictionId} for person ${event.personId} not found in NOMIS")
+      try {
+        val nomisPerson = nomisApiService.getPerson(nomisPersonId = event.personId)
+        val nomisRestriction = nomisPerson.restrictions.find { it.id == event.visitorRestrictionId } ?: throw IllegalStateException("Restriction ${event.visitorRestrictionId} for person ${event.personId} not found in NOMIS")
+
+        mappingApiService.getByNomisPersonRestrictionIdOrNull(nomisPersonRestrictionId = event.visitorRestrictionId)?.also {
+          telemetry["dpsContactRestrictionId"] = it.dpsId
+          dpsApiService.updateContactRestriction(contactRestrictionId = it.dpsId.toLong(), nomisRestriction.toDpsUpdateContactRestrictionRequest(dpsContactId = event.personId))
+          telemetryClient.trackEvent(
+            "$telemetrySuffix-updated-success",
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/contactperson/ContactPersonSynchronisationService.kt:72]: Isn't that a prefix?

Comment (andymarke) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/contactperson/ContactPersonSynchronisationService.kt:72]: Just this above
` val telemetrySuffix = "contactperson-person-restriction-synchronisation"`

I was trying to reduce a bit of duplication; but maybe I'll go back to the full hardcode string. It is perhaps just confusing

**Review by karenmillermoj — APPROVED** [review]
lgtm

**Pull request merged**